decoder/expr_any_for.go

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package decoder
+
+import (
+	"context"
+
+	"github.com/hashicorp/hcl-lang/lang"
+	"github.com/hashicorp/hcl-lang/reference"
+	"github.com/hashicorp/hcl-lang/schema"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func (a Any) completeForExprAtPos(ctx context.Context, pos hcl.Pos) ([]lang.Candidate, bool) {
+	candidates := make([]lang.Candidate, 0)
+
+	switch eType := a.expr.(type) {
+	case *hclsyntax.ForExpr:
```

Comment:
Can we recover the `.` when triggering completion with a prefix?

<img width="863" alt="CleanShot 2024-02-06 at 10 24 57@2x" src="https://github.com/hashicorp/hcl-lang/assets/45985/944ed06e-d342-47c5-9435-4e12f58f474a">

Reply:
That would be a little more difficult here unfortunately as we'd have to be recovering pretty much everything up to the `[` _at least_. I didn't even think of the consequences of the incomplete `ForExpr` wrapped into another expression like function here but I expect it to be similarly difficult.

Either way we now track it under https://github.com/hashicorp/hcl-lang/issues/370